lxd/identities.go

```diff
@@ -433,6 +434,169 @@ func getIdentity(d *Daemon, r *http.Request) response.Response {
+		if err != nil {
+			return fmt.Errorf("Failed to get current identity from database: %w", err)
+		}
+
+		apiIdentity, err = id.ToAPI(ctx, tx.Tx())
+		if err != nil {
+			return fmt.Errorf("Failed to populate LXD groups: %w", err)
+		}
+
+		effectiveGroups = apiIdentity.Groups
+		if len(identityProviderGroups) > 0 {
+			var args []any
+			for _, idpGroup := range identityProviderGroups {
+				args = append(args, idpGroup)
+			}
+
+			q := fmt.Sprintf(`
+SELECT DISTINCT auth_groups.name
+FROM auth_groups
+JOIN auth_groups_identity_provider_groups ON auth_groups.id = auth_groups_identity_provider_groups.auth_group_id
+JOIN identity_provider_groups ON auth_groups_identity_provider_groups.identity_provider_group_id = identity_provider_groups.id
+WHERE identity_provider_groups.name IN %s`, query.Params(len(identityProviderGroups)))
+			mappedGroups, err := query.SelectStrings(ctx, tx.Tx(), q, args...)
+			if err != nil {
+				return fmt.Errorf("Failed to get identity provider group mappings: %w", err)
+			}
+
+			for _, mappedGroup := range mappedGroups {
+				if !shared.ValueInSlice(mappedGroup, effectiveGroups) {
+					effectiveGroups = append(effectiveGroups, mappedGroup)
+				}
+			}
+		}
+
+		if len(effectiveGroups) > 0 {
+			var args []any
+			for _, effectiveGroup := range effectiveGroups {
+				args = append(args, effectiveGroup)
+			}
+
+			q := fmt.Sprintf(`
+SELECT DISTINCT auth_groups_permissions.entitlement, auth_groups_permissions.entity_type, auth_groups_permissions.entity_id
+FROM auth_groups_permissions
+JOIN auth_groups ON auth_groups_permissions.auth_group_id = auth_groups.id
+WHERE auth_groups.name IN %s`, query.Params(len(effectiveGroups)))
+
+			rows, err := tx.Tx().QueryContext(ctx, q, args...)
+			if err != nil {
+				return fmt.Errorf("Failed to query effective permissions: %w", err)
+			}
+
+			var permissions []dbCluster.Permission
+			for rows.Next() {
+				var permission dbCluster.Permission
+				err := rows.Scan(&permission.Entitlement, &permission.EntityType, &permission.EntityID)
+				if err != nil {
+					return fmt.Errorf("Failed to scan effective permissions: %w", err)
+				}
+
+				permissions = append(permissions, permission)
+			}
+
+			permissions, danglingPermissions, entityURLs, err := dbCluster.GetPermissionEntityURLs(ctx, tx.Tx(), permissions)
+			if err != nil {
+				return fmt.Errorf("Failed to get entity URLs for effective permissions: %w", err)
+			}
+
+			if len(danglingPermissions) > 0 {
+				permissionIDs := make([]int, 0, len(danglingPermissions))
+				entityTypes := make([]dbCluster.EntityType, 0, len(danglingPermissions))
+				for _, perm := range danglingPermissions {
+					permissionIDs = append(permissionIDs, perm.ID)
+					if !shared.ValueInSlice(perm.EntityType, entityTypes) {
+						entityTypes = append(entityTypes, perm.EntityType)
+					}
+				}
+
+				logger.Warn("Encountered dangling permissions", logger.Ctx{"permission_ids": permissionIDs, "entity_types": entityTypes})
```

Comment:
Now we aren't having to delete dangling permissions returned from dbCluster.GetPermissionEntityURLs, it would seem to make sense to not return them at all and instead handle the logging of detected dangling perms inside dbCluster.GetPermissionEntityURLs, rather than relying on every call site to do it.